Add mention of new `env.d.ts` file to TypeScript page (#1202)
withastro/docs: src/pages/en/guides/typescript.md

    "resolveJsonModule": true,
    // Enable stricter transpilation for better output.
    "isolatedModules": true,
    // Astro will directly run your TypeScript code, no transpilation needed.
    "noEmit": true
  }
}
```

> Additionally, our templates include an `env.d.ts` file inside the `src` folder to provide [Vite's client types](https://vitejs.dev/guide/features.html#client-types) to your project:

Author: Is it "an `env.d.ts` file" or "a `env.d.ts` file"?

Reviewer: I would pretty confidently say "an" for English if it's not written with a "dot" before the env. It's mostly when files start with a dot then a vowel that it can get tricky.  😅

    "resolveJsonModule": true,
    // Enable stricter transpilation for better output.
    "isolatedModules": true,
    // Astro will directly run your TypeScript code, no transpilation needed.
    "noEmit": true
  }
}
```

> Additionally, our templates include `env.d.ts` file inside the `src` folder that contains the following:
> 
> ```typescript title="env.d.ts"
> /// <reference types="astro/client" />
> ```
> 
> This will includes [Vite's client types](https://vitejs.dev/guide/features.html#client-types) inside your project. Alternatively, if you prefer not having this file in your project, you can delete it and use the [`types` setting](https://www.typescriptlang.org/tsconfig#types) inside your `tsconfig.json` to add them:
> 
> ```json title="tsconfig.json"
> {
>   "compilerOptions": {
>     "types": ["astro/client"]
>   }
> }
> ```

Reviewer: ```suggestion
Additionally, our templates include an `env.d.ts` file inside the `src` folder to provide [Vite's client types](https://vitejs.dev/guide/features.html#client-types) to your project:

```typescript title="env.d.ts"
/// <reference types="astro/client" />
```
Optionally, you can delete this file and instead add the [`types` setting](https://www.typescriptlang.org/tsconfig#types) to your `tsconfig.json`:

```json title="tsconfig.json"
{
  "compilerOptions": {
    "types": ["astro/client"]
  }
}
```
```

Reviewer: Hi @Princesseuh! Just a reminder that this is still here, waiting for you when you're ready!
